connectors/utils.py

```diff
@@ -401,3 +401,26 @@ async def func_to_execute(*args, **kwargs):
 
 def e2str(entry):
     return entry.name.lower()
+
+
+class MultiService:
+    def __init__(self, services):
+        self._services = services
+
+    async def run(self):
```

Comment:
do you want the function to return a list of done, pending or some kind of way to know that one died

this will be useful to return the correct exit code or re-raise an error

Reply:
I can return either stats or an exit code or raise an exception if any exited due to an exception, what would be the best in your opinion?

I feel like raising an exception is the simplest.

Reply:
> I feel like raising an exception is the simplest.
sounds good!

Reply:
Hmmm it's actually not that simple looks like. It's really hard to distinguish the situation of "service died" vs "service stopped cause of Cancellation and gracefully handled it". It probably will require extending service classes to store some metadata about their run - properties like "failed", "cancelled" and so so that we could ask the services whether they ran successfully or not.

Shall we discuss it more in a different PR?

Reply:
>  It's really hard to distinguish the situation of "service died" vs "service stopped cause of Cancellation and gracefully handled it"

unless we specifically ask the Service class to handle `CancelledError` like in https://github.com/elastic/connectors-python/pull/437#discussion_r1103153507 and have a `canceled` flag

Reply:
Indeed, that's what I meant but it feels like it's better to handle it in a separate PR, what do you think?

Reply:
absolutely! small PRs FTW